database/pgx/v5/pgx.go

```diff
@@ -0,0 +1,486 @@
+//go:build go1.9
+// +build go1.9
+
+package pgx
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"io"
+	nurl "net/url"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"go.uber.org/atomic"
+
+	"github.com/golang-migrate/migrate/v4"
+	"github.com/golang-migrate/migrate/v4/database"
+	"github.com/golang-migrate/migrate/v4/database/multistmt"
+	"github.com/hashicorp/go-multierror"
+	"github.com/jackc/pgerrcode"
+	"github.com/jackc/pgx/v5/pgconn"
+	_ "github.com/jackc/pgx/v5/stdlib"
+)
+
+func init() {
+	db := Postgres{}
+	database.Register("pgx", &db)
```

Comment:
This should be registered as `pgx5` to avoid a conflict with `pgx` (v4)

Reply:
As of a few weeks ago, pgx registers it’s database/sql Driver under `pgx/v4` or `pgx/v5`, as well as conditionally registering `pgx` if it’s not already registered by importing another version (see https://github.com/jackc/pgx/blob/9ae852eb583d2dced83b1d2ffe1c8803dda2c92e/stdlib/sql.go#L94). I’m happy to copy in the conflict-avoidance code or if you prefer just leave it out and have v4 register as pgx and pgx/v4, and v5 only register pgx/v5

I definitely think it would be better to match pgx’s `pgx/vN` registration string rather than inventing a new one. I know I didn’t use it in this PR originally but the upstream change is very new!